src/ansys/fluent/core/workflow.py

```diff
@@ -555,15 +559,13 @@ def set_state(self, args: dict) -> None:
         ----------
         args : dict
             State of the arguments.
+
+        Raises
+        ------
+        Exception
+            If operation fails.
         """
-        if self._dynamic_interface:
-            self.get_state()
-            camel_args = {}
-            for key, val in args.items():
-                camel_args[self._snake_to_camel_map[key]] = val
-            self._task.Arguments.set_state(camel_args)
-        else:
-            self._task.Arguments.set_state(args)
+        self._assign(args, "set_state")
```

Comment:
This is a bit ugly but the prime alternative was to use a lambda. I don't want to pass in the whole method directly to avoid triggering any code prematurely.